instance/models/mixins/database.py

```diff
@@ -69,28 +108,55 @@ def provision_mysql(self):
         """
         if settings.INSTANCE_MYSQL_URL_OBJ and not self.mysql_provisioned:
             cursor, connection = _get_mysql_cursor()
-            for database in self.mysql_database_names:
+
+            # Create migration user
+            _create_user(cursor, self.migrate_user, self._get_mysql_pass(self.migrate_user))
+
+            # Create default databases and users, and grant privileges
+            for database in self.mysql_databases:
                 # We can't use the database name in a parameterized query, the
-                # driver doesn't escape it properly. Se we escape it here instead
-                database_name = connection.escape_string(database).decode()
-                cursor.execute('CREATE DATABASE `{0}` DEFAULT CHARACTER SET utf8'.format(database_name))
-                cursor.execute('CREATE USER %s IDENTIFIED BY %s', (self.mysql_user, self.mysql_pass,))
-                cursor.execute('GRANT ALL ON `{0}`.* TO %s'.format(database_name), (self.mysql_user,))
+                # driver doesn't escape it properly. So we escape it here instead
+                database_name = connection.escape_string(database["name"]).decode()
```

Comment:
Now that we have database creation neatly encapsulated inside `_create_database`, we should make that method take care of escaping the database name. I would move the line above into `_create_database` and pass it the raw (unescaped) named here.